### ISSUE TITLE
Just insert formatted reference?

### ISSUE THREAD

**Issue by achamess** [hans/obsidian-citation-plugin#28]
Great plugin in. Thanks for making it. 
I use Paper/ReadCube. I was able to export my references and then convert to CSL-JSOn using Citation.js. 

I can access the references now in Obsidian, but rather than create new literature notes, I'd like to be able to put the formatted reference inside my Obsidian notes. Right now, I can just get the citekey. But how would that eventually render the full citation? Pandoc?

I see one can use variables in templates, but that necessitates the creation of a new note, no?

**Comment by hans** [comment]
You're correct that the new ["Insert Markdown citation" command](https://github.com/hans/obsidian-citation-plugin/blob/master/README.md#usage) can give you half of what you want. But we don't support inserting fully formatted citations at the moment (e.g. in proper APA style).

This feature is on the radar since you and several others have suggested it, but I can't guarantee it'll be available anytime soon. 

For my own reference -- supporting this will require some significant work, either

- linking to a Zotero translator server to get the full formatted citation (complicated for security reasons, because Zotero server doesn't play well with Obsidian due to CORS issues)
  - need to either ask Obsidian devs to relax CORS security, or find a workaround on the Zotero side. I've given the latter a few attempts but nothing has worked yet. The only remaining option I'm aware of which I haven't tried is using a standalone Zotero translation server.
- re-implementing the whole formatting logic on the Obsidian side (probably not a good idea)
- including `citeproc.js` as a dependency and using this (will probably require retooling `Entry` representation)

**Comment by hans** [comment]
Also a prerequisite for #22 .

**Comment by alltagsverstand** [comment]
Just saw this feature request now... I have proposed actually the same functionality in the [obsidian forum](https://forum.obsidian.md/t/new-plugin-citations-with-zotero/9793/89?u=alltagsverstand). 

Because this feature seems to be more complicated, maybe the template-and-shortcut-solution I proposed in the forum might be a temporary workaround?

**Comment by psyguy** [comment]
I was just opening a new issue for this feature. After reading @hans comment above. I figured a quick workaround for *APA-like* bibliography. Here's what I've done:

# Solution

I added the following snippet to the template that kinda mimics APA style (or, what I would have liked APA style to look like):

``` md
###### Bibliography

> {{authorString}} ([[{{year}}]]) **[[{{citekey}}.pdf|{{title}}]]**.
*{{containerTitle}}* | doi:[{{DOI}}](https://doi.org/{{DOI}}) | @{{citekey}}
^apalike

```

My PDFs are renamed to their `citekey`s using [ZotFile](http://zotfile.com/), and like publication years to be links. (If I understand how #41 is fixed, the authors will be liked as well.)

This adds a subheader containing APA-like bibliography as a blockquote, which can be referenced by the `^apalike`.

Then, I set markdown primary citation template to

```
![[@{{citekey}}#^apalike]]
```

Which will embed the blockquote when I want to insert it in other notes.

# Example 

As an example, this `bibtex` entry:

``` bib

@article{asparouhov_2018_DynamicStructuralEquation,
  title = {Dynamic {{Structural Equation Models}}},
  author = {Asparouhov, Tihomir and Hamaker, Ellen L. and Muthén, Bengt},
  date = {2018-05-04},
  journaltitle = {Structural Equation Modeling: A Multidisciplinary Journal},
  volume = {25},
  pages = {359--388},
  publisher = {{Routledge}},
  issn = {1070-5511},
  doi = {10.1080/10705511.2017.1406803},
  url = {https://doi.org/10.1080/10705511.2017.1406803},
  urldate = {2020-10-21},
  keywords = {Baysian methods,dynamic factor analysis,intensive longitudinal data,time series analysis},
  number = {3}
}

```

makes the following note (the template is more than what posted above):

![image](https://user-images.githubusercontent.com/8527082/104920439-1197e380-5998-11eb-9236-39a512b23ed3.png)

And will be embedded in another note as:

![image](https://user-images.githubusercontent.com/8527082/104920627-5754ac00-5998-11eb-8d68-9d4acb7a60e3.png)

(It would be awesome if #39 is resolved.)

---

## Problem 

This workaround is not optimal if some entries (handles?) are missing. For instance, the bibliography for this entry:

``` bib

@book{macdonald1997hidden,
  title={Hidden Markov and other models for discrete-valued time series},
  author={MacDonald, Iain L and Zucchini, Walter},
  volume={110},
  year={1997},
  publisher={CRC Press}
}
```

Gives this:

![image](https://user-images.githubusercontent.com/8527082/104921236-540df000-5999-11eb-9660-d442c382a74f.png)
